examples/adaptiveprecision-blockjacobi/adaptiveprecision-blockjacobi.cpp

```diff
@@ -93,11 +92,11 @@ int main(int argc, char *argv[])
 
     // copy b again
     b->copy_from(host_x.get());
-
+    ValueType reduction_factor = 1e-7;
```

Comment:
Like I said in a comment, [`std::numeric_limit<ValueType>::epsilon`](https://en.cppreference.com/w/cpp/types/numeric_limits/epsilon) should be more suitable here (hard-coding these in my opinion is bad practice):
```suggestion
    ValueType reduction_factor = std::numeric_limit<ValueType>::epsilon;
```

Reply:
Also, `reduction_factor` should be `const` everywhere.

Reply:
I dont think using `numeric_limits<>::epsilon()` makes sense here, because that says, atleast to me that the reduction_factor has to be the epsilon of that type, which is not true. 1e-7 makes perfect sense even if you are using just `double` . And it is definitely something the user would like to tweak.